### ISSUE TITLE
Lint/Void should detect `Kernel.tap` as void-context-method

### ISSUE THREAD

**Issue by untitaker** [bbatsov/rubocop#5512]
`Kernel.tap` ignores the return value of the passed block. Rubocop should know about this.

The following code should emit a linting error, since it was clearly the programmer's intent to write a program that outputs `7`:

```ruby
a = 4.tap do |x|
  x = x + 3
  x
end

puts a
```

However, the **actual behavior** is that Rubocop (from master) is completely fine with that code.

I think the fix is trivial, since rubocop does analysis like this for `each` already. Will open a PR soon
